make.config.in

```diff
@@ -90,14 +90,22 @@ libfast: $(BOUT_TOP)/config.status $(BOUT_TOP)/make.config $(DIRS)
 # Recursively run make through subdirs
 ####################################################################
 
-ifneq ("$(TARGET)" , "libfast")
+ifeq ("$(TARGET)" , "lib")
 $(DIRS):
 	@#echo make $@ slowly TARGET $(TARGET)
 	@$(MAKE) -s --no-print-directory TARGET=lib -C $@ all
 else
+ifeq ("$(TARGET)","libfast")
 $(DIRS):
 	@#echo make $@ fast
 	@$(MAKE) -s --no-print-directory TARGET=$(TARGET) -C $@ $(TARGET)
+else
+ifndef MODULE_DIR
+MODULE_DIR=$(shell pwd)
+endif
+$(DIRS):
+	@$(MAKE) -s --no-print-directory MODULE_DIR=$(MODULE_DIR) SUB_NAME=$(shell f=$@ ; g=$${f%/} ; echo $${g##*/}) TARGET=sub -C $@
```

Comment:
Please could you add a comment here? "Strip trailing slash" or something. Also, why it's necessary -- makefiles are already full of black magic :)